packages/x-charts/src/utils/timeTicks.ts

```diff
@@ -0,0 +1,58 @@
+import { TicksFrequency, TicksFrequencyDefinition } from '../models/timeTicks';
+
+function yearNumber(from: Date, to: Date) {
+  return Math.abs(to.getFullYear() - from.getFullYear());
+}
+function monthNumber(from: Date, to: Date) {
+  return Math.abs(
+    to.getFullYear() * 12 + to.getMonth() - 12 * from.getFullYear() - from.getMonth(),
+  );
+}
+function dayNumber(from: Date, to: Date) {
+  return Math.abs(to.getTime() - from.getTime()) / (1000 * 60 * 60 * 24);
+}
+function hourNumber(from: Date, to: Date) {
+  return Math.abs(to.getTime() - from.getTime()) / (1000 * 60 * 60);
+}
+
+export const tickFrequencies: Record<TicksFrequency, TicksFrequencyDefinition> = {
+  years: {
+    getTickNumber: yearNumber,
+    isTick: (prev: Date, value: Date) => value.getFullYear() !== prev.getFullYear(),
+    format: (d: Date) => d.getFullYear().toString(),
+  },
+  quarterly: {
+    getTickNumber: (from: Date, to: Date) => Math.floor(monthNumber(from, to) / 3),
+    isTick: (prev: Date, value: Date) =>
+      value.getMonth() !== prev.getMonth() && value.getMonth() % 3 === 0,
+    format: new Intl.DateTimeFormat('default', { month: 'short' }).format,
```

Comment:
Shouldn't the locale be configurable?

Reply:
Donce :) 

https://github.com/mui/mui-x/issues/20512